README.md

```diff
@@ -41,231 +41,383 @@ This project contains two separate providers that operate in tandem:
 ```
 
-### Configuration
+You can then use this kubeconfig file to access the cluster:
 
-The following environment variables can be overridden for the bootstrap provider:
+```bash 
+export KUBECONFIG=kubeconfig
+kubectl get nodes
+```
 
-| Environment Variable | Description | Default |
-|-----------------------| --------------| --------|
-| `USE_INTERNAL_IMAGE_URL` | Enables the bootstrap controller to use the internal IP of assisted-image-service. The internal IP is used in place of the default URL of the live ISO provided by assisted-service to boot hosts. | `"false"`| 
-| `IMAGE_SERVICE_NAME` | Name of the Service CR for assisted-image-service. This contains the internal IP of the assisted-image-service | `assisted-image-service` |
-| `IMAGE_SERVICE_NAMESPACE` | Namespace that the assisted-image-service's Service CR is in | Defaults to the namespace the bootstrap provider is running in if unset|
+# Developper guide
 
+## Architecture Design
 
-### To Uninstall
-**Delete the instances (CRs) from the cluster:**
+TODO: Outdated, need to update
 
-```sh
-kubectl delete -k config/samples/
-```
+[Detailed architecture design](./docs/architecture_design.md)
 
-**Delete the APIs(CRDs) from the cluster:**
 
-```sh
-make uninstall
-```
+## Using the Makefile
 
-**UnDeploy the controller from the cluster:**
+The Makefile can be used through a distrobox environment to ensure all dependencies are met.
 
 ```sh
-make undeploy
+podman build -f Dockerfile.distrobox -t capoa-build .
+distrobox create --image capoa-build:latest capoa-build
+distrobox enter capoa-build
+
+make <target>
 ```
 
-## Project Distribution
+An exception to this is the `docker-build` target, which should be executed from a system where docker/podman is available (not within distrobox).
 
-Following are the steps to build the installer and distribute this project to users.
+### Building Docker Images
 
-1. Build the installer for the image built and published in the registry:
+The project uses a templated approach to generate Dockerfiles for both bootstrap and controlplane providers from a single `Dockerfile.j2` template.
 
+1. Generate the Dockerfiles:
 ```sh
-make build-installer IMG=<some-registry>/cluster-api-agent:tag
+make generate-dockerfiles
 ```
+This will create:
+* Dockerfile.bootstrap-provider for the bootstrap provider
+* Dockerfile.controlplane-provider for the controlplane provider
 
-NOTE: The makefile target mentioned above generates an 'install.yaml'
-file in the dist directory. This file contains all the resources built
-with Kustomize, which are necessary to install this project without
-its dependencies.
-
-2. Using the installer
-
-Users can just run kubectl apply -f <URL for YAML BUNDLE> to install the project, i.e.:
-
+2. Build the Docker images:
 ```sh
-kubectl apply -f https://raw.githubusercontent.com/<org>/cluster-api-agent/<tag or branch>/dist/install.yaml
+make docker-build-all
 ```
-## Development
 
 ### E2E testing
```

Comment:
the ansible env vars & description can be put in a table or itemized to easier reading; the description of DIST_DIR is not clear; is `dist_dir` actually used?

Reply:
good point, I've added https://issues.redhat.com/browse/MGMT-20653 to tackle this issue at a later stage, as this PR is focusing on usage and we'd have much to improve in the Development section